Plutarch/Lift.hs

```diff
@@ -105,20 +106,27 @@ data LiftError
   | LiftError_CompilationError Text
   deriving stock (Eq)
 
-{- | Convert a Plutarch term to the associated Haskell value. Fail otherwise.
-This will fully evaluate the arbitrary closed expression, and convert the resulting value.
+{- | Like `plift'` but provides trace from evaluation.
+
+ @since 1.2.1
 -}
-plift' :: forall p. PUnsafeLiftDecl p => Config -> ClosedTerm p -> Either LiftError (PLifted p)
-plift' config prog = case compile config prog of
-  Left msg -> Left $ LiftError_CompilationError msg
+pliftTrace :: forall p. PUnsafeLiftDecl p => Config -> ClosedTerm p -> (Either LiftError (PLifted p), [Text])
```

Comment:
Is there any other information we are not returning? We should maybe also call this something else. `pliftMaybe`? The current name is not that bad though.

Reply:
We can return cek budgets, but I don't think that'd be useful. If traces are the only thing we are returning alongside, `pliftTrace` would be a better name.

Reply:
We should return the budget, because someone will come along in the future, and make another function that returns the budget.